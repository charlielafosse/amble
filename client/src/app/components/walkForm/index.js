import React from "react";
import("./style.css");

export const WalkForm = ({ state, handleChange, handleSubmit }) => (
  <form onSubmit={handleSubmit}>
    <h1>Add a walk</h1>
    <label htmlFor="location-of-walk">Location</label>
    <input
      pattern="^[^<>]+$"
      name="location"
      type="text"
      id="location-of-walk"
      placeholder="Where did you walk?"
      required
      onChange={handleChange}
    />
    <br />
    <label htmlFor="difficulty">Difficulty</label>
    <select onChange={handleChange} name="difficulty" id="difficulty">
      <option value="Stroll">Stroll</option>
      <option value="Easy">Easy</option>
      <option value="Medium">Medium</option>
      <option value="Brisk">Brisk</option>
      <option value="Exhausting">Exhausting</option>
    </select>
    <br />
    <button type="submit">Submit</button>
  </form>
);
