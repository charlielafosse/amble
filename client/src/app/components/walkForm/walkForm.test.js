import React from "react";
import renderer from "react-test-renderer";
import { WalkForm } from "../walkForm";

test("WalkForm renders properly", () => {
  const component = renderer.create(<WalkForm />).toJSON();
  expect(component).toMatchSnapshot();
});
