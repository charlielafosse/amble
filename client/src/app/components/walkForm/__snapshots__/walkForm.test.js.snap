// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalkForm renders properly 1`] = `
<form>
  <h1>
    Add a walk
  </h1>
  <label
    htmlFor="location-of-walk"
  >
    Location
  </label>
  <input
    id="location-of-walk"
    name="location"
    pattern="^[^<>]+$"
    placeholder="Where did you walk?"
    required={true}
    type="text"
  />
  <br />
  <label
    htmlFor="difficulty"
  >
    Difficulty
  </label>
  <select
    id="difficulty"
    name="difficulty"
  >
    <option
      value="Stroll"
    >
      Stroll
    </option>
    <option
      value="Easy"
    >
      Easy
    </option>
    <option
      value="Medium"
    >
      Medium
    </option>
    <option
      value="Brisk"
    >
      Brisk
    </option>
    <option
      value="Exhausting"
    >
      Exhausting
    </option>
  </select>
  <br />
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
